services/motion/builtin/builtin.go

```diff
@@ -327,7 +338,22 @@ func (ms *builtIn) MoveOnGlobe(
 			if resp.err != nil {
 				return false, resp.err
 			}
+			ms.logger.Info("obstacle detection triggering a replan")
+		}
+
+		if maxReplans >= 0 {
+			i++
+			if i > maxReplans {
+				return false, fmt.Errorf("exceeded maximum number of replans: %d", maxReplans)
+			}
+		}
+		lastPlan := mr.seedPlan
+		// TODO: RSDK-4509 obstacles should include any transient obstacles which may have triggered a replan, if any.
+		mr, err = ms.newMoveOnGlobeRequest(ctx, componentName, destination, movementSensorName, obstacles, motionCfg, extra)
+		if err != nil {
+			return false, err
 		}
+		mr.seedPlan = lastPlan
```

Comment:
why do we need to set the seed plan back to what it was before after calling `newMoveOnGlobeRequest`?

Reply:
You're right, this is a little odd; this should be passed to `newMoveOnGlobeRequest`